src/sys/expressionparser.cxx

```diff
@@ -219,11 +219,13 @@ std::shared_ptr<FieldGenerator> ExpressionParser::parseParenExpr(LexInfo &lex) {
   lex.nextToken(); // eat '('
   
   std::shared_ptr<FieldGenerator> g = parseExpression(lex);
-  if(!g)
-    return NULL;
-  
-  if((lex.curtok != ')') && (lex.curtok != ']'))
-    return NULL;
+  if (!g)
+    throw ParseException("Incomplete parentheses; did you forget a ')'?");
```

Comment:
I think we can just delete this completely, as `parseExpression` now returns something or throws

Reply:
Ah good spot. I think I deleted the wrong if; the check
`if((lex.curtok != ')') && (lex.curtok != ']'))` 
should be there, rather than `if (!g)`